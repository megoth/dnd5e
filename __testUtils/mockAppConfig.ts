export default function mockAppConfig() {
  return {
    errorsUrl: "http://localhost:3000/data/errors.ttl",
    translationsUrl: "http://localhost:3000/data/translations.ttl",
  };
}
