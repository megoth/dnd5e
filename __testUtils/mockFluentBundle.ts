import { FluentBundle, FluentResource } from "@fluent/bundle";
import { getTranslationId } from "../src/models/translation";
import { defaultLocale } from "./mockLanguage";

interface Options {
  locale: string;
}

export default function mockFluentBundle(
  translations: Record<string, string> = {},
  options: Partial<Options> = {}
) {
  const locale = options.locale || defaultLocale;
  const bundle = new FluentBundle(locale);
  Object.entries(translations).forEach(([translationURL, message]) => {
    const translationId = getTranslationId(translationURL);
    bundle.addResource(new FluentResource(`${translationId} = ${message}`));
  });
  return bundle;
}
