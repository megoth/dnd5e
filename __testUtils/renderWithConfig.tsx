import React from "react";
import { render } from "@testing-library/react";
import AppConfigProvider from "../src/contexts/appConfig";
import mockAppConfig from "./mockAppConfig";

export default function renderWithConfig(ui) {
  const { errorsUrl, translationsUrl } = mockAppConfig();
  return render(
    <AppConfigProvider errorsUrl={errorsUrl} translationsUrl={translationsUrl}>
      {ui}
    </AppConfigProvider>
  );
}
