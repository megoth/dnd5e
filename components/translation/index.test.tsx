import { LocalizationProvider, ReactLocalization } from "@fluent/react";
import React from "react";
import Translation from "./index";
import mockFluentBundle from "../../__testUtils/mockFluentBundle";
import renderWithConfig from "../../__testUtils/renderWithConfig";
import mockAppConfig from "../../__testUtils/mockAppConfig";

describe("Translation", () => {
  it("renders translated message", () => {
    const { translationsUrl } = mockAppConfig();
    const id = "test";
    const message = `This is a test`;
    const bundle = mockFluentBundle(translationsUrl, {
      [id]: message,
    });
    const l10n = new ReactLocalization([bundle]);
    const { asFragment, getByText } = renderWithConfig(
      <LocalizationProvider l10n={l10n}>
        <Translation id={id} />
      </LocalizationProvider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(message)).toBeDefined();
  });
});
