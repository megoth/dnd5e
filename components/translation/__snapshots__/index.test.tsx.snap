// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Translation renders translated message 1`] = `
<DocumentFragment>
  This is a test
</DocumentFragment>
`;
